app/apps.py

```diff
@@ -1,60 +1,65 @@
 import json
 import re
 
-from . import db, utils
+from . import db, search, utils
 
 
-def load_appstream():
+def load_appstream(app_downloads):
     apps = utils.appstream2dict("repo")
 
     current_apps = {app[5:] for app in db.redis_conn.smembers("apps:index")}
     current_categories = db.redis_conn.smembers("categories:index")
     current_developers = db.redis_conn.smembers("developers:index")
 
-    db.initialize()
     with db.redis_conn.pipeline() as p:
         p.delete("categories:index", *current_categories)
         p.delete("developers:index", *current_developers)
 
+        clean_html_re = re.compile("<.*?>")
+        search_apps = []
         for appid in apps:
             redis_key = f"apps:{appid}"
 
-            clean_html_re = re.compile("<.*?>")
             search_description = re.sub(clean_html_re, "", apps[appid]["description"])
 
-            if search_keywords := apps[appid].get("keywords"):
-                search_keywords = " ".join(search_keywords)
-            else:
-                search_keywords = ""
-
-            fts = {
-                "id": appid,
-                "name": apps[appid]["name"],
-                "summary": apps[appid]["summary"],
-                "description": search_description,
-                "keywords": search_keywords,
-            }
+            search_keywords = apps[appid].get("keywords")
+
+            # order of the dict is important for attritbute ranking
+            search_apps.append(
+                {
+                    "id": utils.get_clean_app_id(appid),
+                    "name": apps[appid]["name"],
+                    "summary": apps[appid]["summary"],
+                    "downloads_last_month": app_downloads.get(appid, {'downloads_last_month': 0}).get('downloads_last_month', 0),
```

Comment:
`app_downloads` has everything we have stats for, but we only want items, that are in the appstream to get pushed to the search (e.g. no EOL apps)